TailLight/device.cpp

```diff
@@ -1,46 +1,70 @@
 #include "driver.h"
 #include <Hidport.h>
+#include "SetBlack.h"
 
 EVT_WDF_IO_QUEUE_IO_DEVICE_CONTROL EvtIoDeviceControlFilter;
 
 
-VOID EvtSetBlackTimer(_In_ WDFTIMER  Timer) {
-    KdPrint(("TailLight: EvtSetBlackTimer begin\n"));
+NTSTATUS PnpNotifyDeviceInterfaceChange(
+    _In_ PVOID pvNotificationStructure,
+    _Inout_opt_ PVOID pvContext) {
 
-    WDFDEVICE device = (WDFDEVICE)WdfTimerGetParentObject(Timer);
-    NT_ASSERTMSG("EvtSetBlackTimer device NULL\n", device);
+    //KdPrint(("TailLight: PnpNotifyDeviceInterfaceChange enter\n"));
+    NT_ASSERTMSG("WDFDEVICE not passed in!", pvContext);
 
-    NTSTATUS status = SetFeatureColor(device, 0);
-    if (!NT_SUCCESS(status)) {
-        KdPrint(("TailLight: EvtSetBlackTimer failure NTSTATUS=0x%x\n", status));
-        return;
+    if (pvNotificationStructure == NULL) {
+        return STATUS_SUCCESS;
+    }
+
+    PDEVICE_INTERFACE_CHANGE_NOTIFICATION pDevInterface =
+        (PDEVICE_INTERFACE_CHANGE_NOTIFICATION)pvNotificationStructure;
+
+    ASSERT(IsEqualGUID(*(_GUID*)&(pDevInterface->InterfaceClassGuid),
```

Comment:
I don´t think you need the `*(_GUID*)&` casting here. Same also apply to the IsEqualGUID call below.